src/app/components/algorithms/algorithm-publications-list/algorithm-publications-list.component.ts

```diff
@@ -10,42 +14,124 @@ import { EntityModelPublicationDto } from 'api/models/entity-model-publication-d
 })
 export class AlgorithmPublicationsListComponent implements OnInit {
   @Input() algorithm: EntityModelAlgorithmDto;
+  @Input() linkedPublications: EntityModelPublicationDto[] = [];
 
   publications: EntityModelPublicationDto[];
   variableNames: string[] = ['title', 'authors', 'doi'];
   tableColumns: string[] = ['Title', 'Authors', 'DOI'];
-  pagingInfo: any = {};
-  paginatorConfig: any = {
-    amountChoices: [10, 25, 50],
-    selectedAmount: 10,
+  linkObject: any = {
+    title: 'Link publication with ',
+    subtitle: 'Search publications by title',
+    displayVariable: 'title',
+    data: [],
   };
+  tableAddAllowed = true;
+  isLinkingEnabled = false;
 
-  constructor(private algorithmService: AlgorithmService) {}
+  constructor(
+    private algorithmService: AlgorithmService,
+    private publicationService: PublicationService,
+    private router: Router
+  ) {}
 
   ngOnInit(): void {
+    this.linkObject.title += this.algorithm.name;
+    this.getLinkedPublications({ algoId: this.algorithm.id });
+  }
+
+  getLinkedPublications(params): void {
+    this.algorithmService.getPublications(params).subscribe((publications) => {
+      if (publications._embedded) {
+        this.linkedPublications = publications._embedded.publications;
+      } else {
+        this.linkedPublications = [];
+      }
+    });
+  }
+
+  searchUnlinkedPublications(search: string): void {
+    // Search for unlinked algorithms if search-text is not empty
+    if (search) {
+      this.publicationService.getPublications2({ search }).subscribe((data) => {
+        this.updateLinkablePublications(data._embedded);
+      });
+    } else {
+      this.linkObject.data = [];
+    }
+  }
+
+  linkPublication(publication: PublicationDto): void {
+    // Empty unlinked algorithms
+    this.linkObject.data = [];
+    // Link algorithm
     this.algorithmService
-      .getPublications({ algoId: this.algorithm.id })
-      .subscribe(
-        (publications) => {
-          if (publications._embedded) {
-            this.publications = publications._embedded.publications;
-          } else {
-            this.publications = [];
+      .addPublication({ algoId: this.algorithm.id, body: publication })
+      .subscribe((data) => {
+        this.getLinkedPublications({ algoId: this.algorithm.id });
+      });
+  }
+
+  async unlinkPublications(event): Promise<void> {
+    // Iterate all selected algorithms
+    for (const publication of event.elements) {
+      await this.unlinkPublication(publication.id);
+    }
+  }
+
+  unlinkPublication(publicationId) {
+    const promise = new Promise((resolve, reject) => {
```

Comment:
Why create a new Promise?

You can simply do it like this:
```ts
async unlinkPublication(): Promise<void> {
  await this.algorithmService.deleteReferenceToPublication( ... ).toPromise();
  // Update table after deletion
  this.getLinkedPublications({ ...  });
}
```

Reply:
Did not know that, will adjust it later

Reply:
Thanks for the tip, pushed it right now. Works and looks much cleaner then the promise example I found